Adamant/Modules/Chat/View/ChatViewController.swift

```diff
@@ -669,6 +670,11 @@ private extension ChatViewController {
     func updateScrollDownButtonVisibility() {
         scrollDownButton.isHidden = isScrollPositionNearlyTheBottom
     }
+    
+    func updateScrollDownButtonCounter() {
+        let count = viewModel.chatroom?.getUnreadCount() ?? 0
+        scrollDownButton.updateCounter(count: count)
+    }
```

Comment:
I suggest implementing the following logic:
1. User scrolls the collection view.
2. The view controller sends the currently visible cells to a function in the view model.
3. The view model function checks if the status of any message needs to be set to "read" and updates the unread count for this chat.

Use Combine to handle this.

With this logic, the unread message count will decrease as the user scrolls smoothly

Reply:
just to clarify, sends to the view model corresponding indexes, not cells itself